src/vivarium/framework/event.py

```diff
@@ -115,11 +115,26 @@ def emit(self, index: pd.Index[int], user_data: dict[str, Any] | None = None) ->
         """
         if not user_data:
             user_data = {}
+
+        clock = self.manager.clock()
+        step_size = self.manager.step_size()
+        event_time: ClockTime
+        if isinstance(clock, int) and isinstance(step_size, int):
+            event_time = clock + step_size
+        elif isinstance(clock, (pd.Timestamp, datetime)) and isinstance(
```

Comment:
Apparently you can just do datetime and timedelta, since the pandas types inherit from them